code/modules/projectiles/updated_projectiles/gun_system.dm

```diff
@@ -54,6 +54,7 @@
 	var/movement_acc_penalty_mult = 5				//Multiplier. Increased and decreased through attachments. Multiplies the accuracy/scatter penalty of the projectile when firing onehanded while moving.
 
 	var/fire_delay = 0							//For regular shots, how long to wait before firing again.
+	var/autofire_delay = 0.4 SECONDS			//Delay between automatic fire shots.
```

Comment:
Component argument instead of var on the gun

Reply:
This may require a larger refactor, by how the gun values are set.
I'll do that in a different PR.